packages/http-client/src/remote-pin-api.ts

```diff
@@ -1,39 +1,80 @@
 import { fetchJson, PinApi } from '@ceramicnetwork/common'
 import { StreamID } from '@ceramicnetwork/streamid'
 import { PublishOpts } from '@ceramicnetwork/common'
+import { DID } from 'dids'
+import { MissingDIDError } from './utils.js'
 
 /**
  * PinApi for Ceramic HTTP client
  */
 export class RemotePinApi implements PinApi {
-  constructor(private readonly _apiUrl: URL) {}
+  // Stored as a member to make it easier to inject a mock in unit tests
+  private readonly _fetchJson: typeof fetchJson = fetchJson
+
+  constructor(private readonly _apiUrl: URL, private readonly _getDidFn: () => DID) {}
+
+  private async buildJWS(actingDid: DID, code: string, url: URL): Promise<string> {
+    if (!actingDid) throw new MissingDIDError()
+    const jws = await actingDid.createJWS({
+      code: code,
+      requestPath: url.pathname,
+    })
+    return `${jws.signatures[0].protected}.${jws.payload}.${jws.signatures[0].signature}`
+  }
+
+  readonly getCodePath = './admin/getCode'
+
+  private getCodeUrl(): URL {
```

Comment:
Could you also add a TODO here to unify this with `RemoteAdminApi`? Have you considered moving this code there and perhaps use composition to encapsulate the `/models` and `/pins` functionality? I understand that this PR is under time pressure, that's why you choose this approach, right?